### PR TITLE
[doc] update doc & fix typo & fmt

### LINKED ISSUE #161: [Bug] Crash when running unit test the second time
# 🐛 Bug

When you run `move package test` the first time, it is fine.

However, if you run `move package test` again, it fails.

If you run `move package build` and then `move package test`, it is good. 

## To reproduce

I just used the step-2 example in the tutorial. (Can refer to #160)

**Stack trace/error message**
```
RUST_BACKTRACE=1 move package test
CACHED MoveStdlib
BUILDING BasicCoin
thread 'main' panicked at 'called Result::unwrap() on an Err value: No such file or directory './build/MoveStdlib/sources/ASCIITests.move'', language/tools/move-cli/src/package/cli.rs:510:83
stack backtrace:
0: rust_begin_unwind
at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
1: core::panicking::panic_fmt
at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
2: core::result::unwrap_failed
at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/result.rs:1749:5
3: core::ops::function::impls::<impl core::ops::function::FnMut for &mut F>::call_mut
4: move_package::compilation::compiled_package::CompiledPackage::build
5: move_package::compilation::build_plan::BuildPlan::compile_with_driver
6: move_cli:📦:cli::run_move_unit_tests
7: move_cli:📦:cli::handle_package_commands
8: move_cli::run_cli
9: move_cli::move_cli
10: move::main
note: Some details are omitted, run with RUST_BACKTRACE=full for a verbose backtrace.
```

## Expected Behavior

Should be able to run the unit test successfully.

## System information

- MacOS Monterey Version 12.3.1
- rustc 1.56.1

Don't know how to check move version...

### LINKED ISSUE #171: [Bug][tutorial] `move package test --coverage` panicked when following language/documentation/tutorial/step_2
# 🐛 Bug

## To reproduce
follow setup guide in [language/documentation/tutorial/README.md](https://github.com/move-language/move/blob/75fb97eee86966d2c8e6be96fe21da969250e36c/language/documentation/tutorial/README.md) and run `move package test --coverage` at the end of 
step_2

**Code snippet to reproduce**
```rust
move/language/documentation/tutorial/step_2/BasicCoin$ RUST_BACKTRACE=full  move package test --coverage
```

**Stack trace/error message**
```
CACHED MoveStdlib
BUILDING BasicCoin
Running Move unit tests
[ PASS    ] 0xcafe::BasicCoin::test_mint_10
Test result: OK. Total tests: 1; passed: 1; failed: 0
thread 'main' panicked at 'Unable to open coverage trace file '"./.trace"'', /home/jiangying/.cargo/git/checkouts/move-cd543da0a1ae83ac/98ed299/language/tools/move-coverage/src/coverage_map.rs:66:33
stack backtrace:
   0:     0x56133f51d6ad - std::backtrace_rs::backtrace::libunwind::trace::hee598835bc88d35b
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x56133f51d6ad - std::backtrace_rs::backtrace::trace_unsynchronized::h9cdc730ba5cf5d72
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x56133f51d6ad - std::sys_common::backtrace::_print_fmt::h75aeaf7ed30e43fa
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x56133f51d6ad - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h606862f787600875
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x56133f547e7c - core::fmt::write::he803f0f418caf762
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/fmt/mod.rs:1190:17
   5:     0x56133f5179f8 - std::io::Write::write_fmt::h70bc45872f37e7bb
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/io/mod.rs:1657:15
   6:     0x56133f51f8b7 - std::sys_common::backtrace::_print::h64d038cf8ac3e13e
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x56133f51f8b7 - std::sys_common::backtrace::print::h359300b4a7fccf65
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x56133f51f8b7 - std::panicking::default_hook::{{closure}}::hf51be35e2f510149
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:295:22
   9:     0x56133f51f580 - std::panicking::default_hook::h03ca0f22e1d2d25e
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:314:9
  10:     0x56133f520009 - std::panicking::rust_panic_with_hook::h3b7380e99b825b63
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:698:17
  11:     0x56133f51fcf7 - std::panicking::begin_panic_handler::{{closure}}::h8e849d0710154ce0
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:588:13
  12:     0x56133f51db74 - std::sys_common::backtrace::__rust_end_short_backtrace::hedcdaddbd4c46cc5
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:138:18
  13:     0x56133f51fa09 - rust_begin_unwind
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
  14:     0x56133e5b67e3 - core::panicking::panic_fmt::he1bbc7336d49a357
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
  15:     0x56133e5eae52 - move_coverage::coverage_map::CoverageMap::update_coverage_from_trace_file::{{closure}}::h65048bc22f7da425
                               at /home/jiangying/.cargo/git/checkouts/move-cd543da0a1ae83ac/98ed299/language/tools/move-coverage/src/coverage_map.rs:66:33
  16:     0x56133e5eae52 - core::result::Result<T,E>::unwrap_or_else::hf67ce6497848f969
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/result.rs:1428:23
  17:     0x56133e5eae52 - move_coverage::coverage_map::CoverageMap::update_coverage_from_trace_file::hf79fa6b098e3a8ce
                               at /home/jiangying/.cargo/git/checkouts/move-cd543da0a1ae83ac/98ed299/language/tools/move-coverage/src/coverage_map.rs:65:20
  18:     0x56133e5eae52 - move_coverage::coverage_map::CoverageMap::from_trace_file::h3b2d4e544c2a02a6
                               at /home/jiangying/.cargo/git/checkouts/move-cd543da0a1ae83ac/98ed299/language/tools/move-coverage/src/coverage_map.rs:96:9
  19:     0x56133e64d2ff - move_cli::package::cli::run_move_unit_tests::h22a53a26c2158f2d
                               at /home/jiangying/.cargo/git/checkouts/move-cd543da0a1ae83ac/98ed299/language/tools/move-cli/src/package/cli.rs:508:28
  20:     0x56133e64bf2c - move_cli::package::cli::handle_package_commands::hf314bba6a05beb3e
                               at /home/jiangying/.cargo/git/checkouts/move-cd543da0a1ae83ac/98ed299/language/tools/move-cli/src/package/cli.rs:389:26
  21:     0x56133e5d103f - move_cli::run_cli::h268d8fca2bd20fa2
                               at /home/jiangying/.cargo/git/checkouts/move-cd543da0a1ae83ac/98ed299/language/tools/move-cli/src/lib.rs:110:37
  22:     0x56133e5baa89 - df_cli::main::h2400e4a5f30ab4c6
                               at /home/jiangying/code/diem/diem-move/df-cli/src/main.rs:30:37
  23:     0x56133e5bb243 - core::ops::function::FnOnce::call_once::h8af1ceb3b2f45b3d
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:227:5
  24:     0x56133e5bb243 - std::sys_common::backtrace::__rust_begin_short_backtrace::h26afe747b2607aa6
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:122:18
  25:     0x56133e5bb1ad - std::rt::lang_start::{{closure}}::hfbdde224977300b5
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/rt.rs:145:18
  26:     0x56133f51cd91 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hb7014f43484a8b4e
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:259:13
  27:     0x56133f51cd91 - std::panicking::try::do_call::h7bc9dc436daeb8c7
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:492:40
  28:     0x56133f51cd91 - std::panicking::try::h653d68a27ff5f175
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:456:19
  29:     0x56133f51cd91 - std::panic::catch_unwind::h9d739f9f59895e68
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panic.rs:137:14
  30:     0x56133f51cd91 - std::rt::lang_start_internal::{{closure}}::hf006f2bc7ce22bbe
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/rt.rs:128:48
  31:     0x56133f51cd91 - std::panicking::try::do_call::hfb39d6df61a2e69f
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:492:40
  32:     0x56133f51cd91 - std::panicking::try::h13e2d225134958ac
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:456:19
  33:     0x56133f51cd91 - std::panic::catch_unwind::h3bd49b5a5dfb1a50
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panic.rs:137:14
  34:     0x56133f51cd91 - std::rt::lang_start_internal::h2ba92edce36c035e
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/rt.rs:128:20
  35:     0x56133e5bb0e8 - main
  36:     0x7f9316c2cc87 - __libc_start_main
  37:     0x56133e5b706a - _start
  38:                0x0 - <unknown>

```

## Expected Behavior

```
INCLUDING DEPENDENCY MoveStdlib
BUILDING BasicCoin
Running Move unit tests
[ PASS    ] 0xcafe::BasicCoin::test_mint_10
Test result: OK. Total tests: 1; passed: 1; failed: 0
filename is: "./.trace"
```

## System information

- df-cli 0.1.0
- rustc 1.60.0 (7737e0b5c 2022-04-04)
- Ubuntu 18.04.5 LTS x86_64

## Additional context

no

### PR DESCRIPTION
**Pull request by jiangying000**
trying to fix #171

## Motivation

fix bug on tutorial

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

yes

## Test Plan

work through the tutorial after doc change

### CODE REVIEW

**Comment by wrwg** [comment]
There was parallel work on the tutorial in #175 which has landed. Sorry for not paying attention to this PR earlier. Could  you rebase and integrate your changes?

**Comment by djy0** [comment]
> There was parallel work on the tutorial in #175 which has landed. Sorry for not paying attention to this PR earlier. Could  you rebase and integrate your changes?

never mind, I would rebase and continue the work

File: language/documentation/tutorial/step_8_sol/BasicCoin/sources/BasicCoin.move [hunk 1]
```diff
@@ -20,8 +20,8 @@ module NamedAddr::BasicCoin {
 
     public fun publish_balance<CoinType>(account: &signer) {
         let empty_coin = Coin<CoinType> { value: 0 };
-        assert!(!exists<Balance<CoinType>>(signer::address_of(account)), errors::already_published(EALREADY_HAS_BALANCE));
-        move_to(account, Balance<CoinType> { coin:  empty_coin });
+        assert!(!exists<Balance<CoinType>>(Signer::address_of(account)), Errors::already_published(EALREADY_HAS_BALANCE));
```
Comment (sblackshear) [inline language/documentation/tutorial/step_8_sol/BasicCoin/sources/BasicCoin.move:23]:
```suggestion
        assert!(!exists<Balance<CoinType>>(signer::address_of(account)), errors::already_published(EALREADY_HAS_BALANCE));
```

**Review by sblackshear — CHANGES_REQUESTED** [review]
LGTM except for the one bad correction

File: language/documentation/tutorial/step_8_sol/BasicCoin/sources/BasicCoin.move (hunk 1 above)
Comment (jiangying000) [reply language/documentation/tutorial/step_8_sol/BasicCoin/sources/BasicCoin.move]: fixed, thank you

**Review by sblackshear — APPROVED** [review]

**Pull request merged**